From 528c81200b167d317e274e782bf235ae22aee1bd Mon Sep 17 00:00:00 2001
From: Calvin Johnson <calvin.johnson@nxp.com>
Date: Thu, 8 Mar 2018 13:58:38 +0530
Subject: [PATCH 400/767] staging: fsl_ppfe/eth: calculate PFE_PKT_SIZE with
 SKB_DATA_ALIGN

pfe packet size was calculated without considering skb data alignment
and this resulted in jumbo frames crashing kernel when the
cacheline size increased from 64 to 128 bytes with
commit 97303480753e ("arm64: Increase the max granular size").

Modify pfe packet size caclulation to include skb data alignment of
sizeof(struct skb_shared_info).

Signed-off-by: Calvin Johnson <calvin.johnson@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl_ppfe/pfe_hif_lib.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/fsl_ppfe/pfe_hif_lib.h b/drivers/staging/fsl_ppfe/pfe_hif_lib.h
index d48eb14a29b8..08031f1f3117 100644
--- a/drivers/staging/fsl_ppfe/pfe_hif_lib.h
+++ b/drivers/staging/fsl_ppfe/pfe_hif_lib.h
@@ -146,7 +146,7 @@ struct tx_queue_desc {
 #define PFE_BUF_SIZE		2048
 #define PFE_PKT_HEADROOM	128
 
-#define SKB_SHARED_INFO_SIZE   (sizeof(struct skb_shared_info))
+#define SKB_SHARED_INFO_SIZE   SKB_DATA_ALIGN(sizeof(struct skb_shared_info))
 #define PFE_PKT_SIZE		(PFE_BUF_SIZE - PFE_PKT_HEADROOM \
 				 - SKB_SHARED_INFO_SIZE)
 #define MAX_L2_HDR_SIZE		14	/* Not correct for VLAN/PPPoE */
-- 
2.17.0

